controllers/usersignup/usersignup_controller.go

```diff
@@ -756,6 +765,71 @@ func (r *Reconciler) ensureSpaceBinding(
 	return nil
 }
 
+// annotateCaptchaAssessment attempts to annotate the original assessment (if any) to provide feedback so that future scores can be improved
+// the attempt to annotate the previous assessment is a best-effort, we do not want to reconcile again if the request fails because captcha is a separate service
+// and should not have any impact to the UserSignup flows if there are issues with this request
+func (r *Reconciler) annotateCaptchaAssessment(ctx context.Context, userSignup *toolchainv1alpha1.UserSignup, newState string) {
+	logger := log.FromContext(ctx)
+	assessmentID, assessmentIDFound := userSignup.Annotations[toolchainv1alpha1.UserSignupCaptchaAssessmentIDAnnotationKey]
+	if !assessmentIDFound {
+		// there's no assessment ID to annotate
+		return
+	}
+
+	newAssessmentAnnotation := r.shouldAnnotateCaptchaAssessment(userSignup, newState)
+	if newAssessmentAnnotation == recaptchapb.AnnotateAssessmentRequest_ANNOTATION_UNSPECIFIED {
+		// no need to annotate the previous assessment
+		return
+	}
+
+	gctx := context.Background()
+	rclient, err := recaptcha.NewClient(gctx)
+	if err != nil {
+		logger.Error(err, "error creating reCAPTCHA client, cannot annotate assessment")
+		return
+	}
+	defer rclient.Close()
+
+	annotateRequest := &recaptchapb.AnnotateAssessmentRequest{
+		Name:       assessmentID,
+		Annotation: newAssessmentAnnotation,
+	}
+
+	response, err := rclient.AnnotateAssessment(gctx, annotateRequest)
+	if err != nil {
+		logger.Error(err, "error annotating assessment")
+		return
+	}
+
+	newAnnotationName, isFound := recaptchapb.AnnotateAssessmentRequest_Annotation_name[int32(newAssessmentAnnotation)]
+	if isFound {
+		// set the annotated assessment value: FRAUDULENT or LEGITIMATE
+		userSignup.Annotations[toolchainv1alpha1.UserSignupCaptchaAnnotatedAssessmentAnnotationKey] = newAnnotationName
+	}
+	logger.Info("Assessment annotated successfully", "assessment_annotation", newAnnotationName, "response", response.String())
+}
+
+// shouldAnnotateCaptchaAssessment determines whether the previous assessment should be annotated
+// returns LEGITIMATE or FRAUDULENT if the assessment should be annotated
+// returns ANNOTATION_UNSPECIFIED otherwise
+func (r *Reconciler) shouldAnnotateCaptchaAssessment(userSignup *toolchainv1alpha1.UserSignup, newState string) recaptchapb.AnnotateAssessmentRequest_Annotation {
```

Comment:
The function name is a bit misleading, isn't it? It doesn't determine if it should be annotated or not but rather always returns a new/expected assessment annotation, right?

Reply:
I was using `ANNOTATION_UNSPECIFIED` as the base case where we don't actually do the annotation. In the other function we check whether it's `ANNOTATION_UNSPECIFIED` and if so we skip annotating the assessment. But I'm fine to change it if it's confusing. Do you have a suggestion to make it more clear?

Reply:
I renamed the function to `getCaptchaAssessmentAnnotation`, let me know if you think another name would be better